src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -189,40 +207,129 @@ irep_idt generate_java_generic_typet::build_generic_tag(
   const java_class_typet &original_class) const
 {
   std::ostringstream new_tag_buffer;
-  new_tag_buffer << original_class.get_tag();
-  new_tag_buffer << "<";
-  bool first=true;
-  for(const typet &type_argument : existing_generic_type
-    .generic_type_arguments())
+  const std::string &original_class_tag = original_class.get_tag().c_str();
+  auto generic_argument_p =
+    existing_generic_type.generic_type_arguments().begin();
+
+  // if the original class is implicitly generic, add type
+  // arguments for all generic outer classes
+  if(is_java_implicitly_generic_class_type(original_class))
```

Comment:
If an inner class is generic inside an outer class that is generic, is it implicitly generic? My reading of the definition of `is_java_implicitly_generic_class_type` says no - but isn't the below code needed in this case?

Reply:
In that case, the inner class would be both generic (generic types vector contains the parameters introduced by the class itself) and implicitly generic (implicit generic types vector contains all parameters introduced by outer classes).